dds/DCPS/transport/framework/TransportReassembly.cpp

```diff
@@ -261,27 +266,22 @@ TransportReassembly::reassemble_i(const SequenceRange& seqRange,
       data.header_.sequence_.getValue(), OPENDDS_STRING(conv).c_str()));
   }
 
-  for (ExpirationQueue::iterator pos = expiration_queue_.begin(), limit = expiration_queue_.upper_bound(MonotonicTimePoint::now());
-       pos != limit;) {
-    if (Transport_debug_level > 5 && fragments_.count(pos->second)) {
-      ACE_DEBUG((LM_DEBUG, "(%P|%t) DBG:   TransportReassembly::reassemble_i "
-                 "purge expired with %d fragments\n", int(fragments_.size())));
-    }
-    fragments_.erase(pos->second);
-    expiration_queue_.erase(pos++);
-  }
+  const MonotonicTimePoint now = MonotonicTimePoint::now();
+  expire(now);
 
   const FragKey key(data.header_.publication_id_, data.header_.sequence_);
-
   FragInfoMap::iterator iter = fragments_.find(key);
+  const MonotonicTimePoint expiration = now + timeout_;
+
   if (iter == fragments_.end()) {
-    const MonotonicTimePoint expiration = MonotonicTimePoint::now() + TimeDuration(300);
     fragments_[key] = FragInfo(firstFrag, FragRangeList(1, FragRange(seqRange, data)), total_frags, expiration);
     expiration_queue_.insert(std::make_pair(expiration, key));
     // since this is the first fragment we've seen, it can't possibly be done
-    VDBG((LM_DEBUG, "(%P|%t) DBG:   TransportReassembly::reassemble_i "
-          "stored first frag, returning false (incomplete) with %d fragments\n",
-          int(fragments_.size())));
+    if (Transport_debug_level > 5 || transport_debug.log_fragment_storage) {
+      ACE_DEBUG((LM_DEBUG, "(%P|%t) DBG:   TransportReassembly::reassemble_i "
+                 "stored first frag, returning false (incomplete) with %B fragments\n",
+                 fragments_.size()));
+    }
```

Comment:
I can't put a comment there, but is it worth adding a log message for the completed case (line 290)?

Reply:
It wasn't logged since it doesn't change `fragments_`.  Not that it couldn't be...